src/views/signup.vue

```diff
@@ -0,0 +1,116 @@
+<div class="formWrapper">
+    <div class="wrap">
+        <h3 class="head">Signup Form</h3>
+        <p class="headPara">Please enter details to create a new user</p>
+        <form @submit.prevent="formSubmit">
+            <label class="formLabel">Firstname</label>
+            <input type="text" v-model="signupData.firstName" />
+            <div v-if="firstNameError" class="error">{{ firstNameError }}</div>
+            <div v-if="!firstNameError && formSubmitted" class="formLabel error">
+                First name must be at least 4 characters long
+            </div>
+            <label class="formLabel">Lastname</label>
+            <input type="text" v-model="signupData.lastName" />
+            <div v-if="lastNameError" class="error">{{ lastNameError }}</div>
+            <div v-if="!lastNameError && formSubmitted" class="formLabel error">
+                Last name must be at least 4 characters long
+            </div>
+            <label class="formLabel">Password</label>
+            <input type="password" v-model="signupData.password" />
+            <div v-if="!passwordError && formSubmitted" class="formLabel error">
+                password must be contain 8 character
+            </div>
+            <label class="formLabel">Email</label>
+            <input type="email" v-model="signupData.email" />
+            <div v-if="!emailError && formSubmitted" class="formLabel error">
+                Please enter a valid email
+            </div>
+            <div v-if="emailError" class="error">{{ emailError }}</div>
+            <label class="formLabel">Role</label>
+            <input type="text" v-model="signupData.roleId" />
+            <div v-if="roleIdError" class="error">{{ roleIdError }}</div>
+            <div v-if="!roleIdError && formSubmitted" class="formLabel error">
+                Role ID cannot be empty
+            </div>
+            <div class="formCheck">
+                <input type="checkbox" v-model="signupData.term" />
+                <label class="checkboxlabel">ACCEPT TERMS AND CONDITIONS</label>
+            </div>
+            <button type="submit" class="submitBtn">Create an account</button>
+        </form>
+    </div>
+</div>
+</template>
+<script>
+import axios from "axios";
+import router from "@/router";
+export default {
+    name: "signupCard",
+    data() {
+        return {
+            signupData: {
+                firstName: "",
+                lastName: "",
+                password: "",
+                email: "",
+                roleId: "",
+                term: false,
+            },
+            formSubmitted: false,
+            firstNameError: "",
+            lastNameError: "",
+            passwordError: "",
+            emailError: "",
+            roleIdError: "",
+        };
+    },
+    methods: {
+        formSubmit() {
+            axios
+                .post("https://pollapi.innotechteam.in/user/register", this.signupData)
+                .then((response) => {
```

Comment:
also you created the store (redux) but why not using it for login and signup? @mukeshsin